ui-trackoccupancydiagram/src/components/utils.ts

```diff
@@ -46,33 +46,61 @@ export const drawText = ({
   ctx.restore();
 };
 
-export const getTickPattern = (minutes: string) => {
-  let tickPattern: TickPattern = 'MINUTE';
-  switch (minutes) {
-    case '00':
-      tickPattern = 'HOUR';
-      break;
-    case '30':
-      tickPattern = 'HALF_HOUR';
-      break;
-    case '15':
-    case '45':
-      tickPattern = 'QUARTER_HOUR';
-      break;
-    case '05':
-    case '10':
-    case '20':
-    case '25':
-    case '35':
-    case '40':
-    case '50':
-    case '55':
-      tickPattern = 'FIVE_MINUTES';
-      break;
-    default:
-      tickPattern = 'MINUTE';
-      break;
-  }
+export const getTickPattern = (minutes: string): TickPattern => {
+  const tickPatternMap: Record<string, TickPattern> = {
+    '00': 'HOUR',
+    '30': 'HALF_HOUR',
+    '15': 'QUARTER_HOUR',
+    '45': 'QUARTER_HOUR',
+    '05': 'FIVE_MINUTES',
+    '10': 'FIVE_MINUTES',
+    '20': 'FIVE_MINUTES',
+    '25': 'FIVE_MINUTES',
+    '35': 'FIVE_MINUTES',
+    '40': 'FIVE_MINUTES',
+    '50': 'FIVE_MINUTES',
+    '55': 'FIVE_MINUTES',
+  };
+  return tickPatternMap[minutes] ?? 'MINUTE';
+};
+
+export const getLabelMarks = (
+  timeRanges: number[],
+  minT: number,
+  maxT: number,
+  labelLevels: number[]
+) => {
+  const labelMarks: Record<number, { level: number; rangeIndex: number }> = {};
+
+  timeRanges.map((range, i) => {
+    const labelLevel = labelLevels[i];
+
+    if (!labelLevel) return;
+
+    let t = Math.floor(minT / range) * range;
+    while (t <= maxT) {
+      if (labelLevel) labelMarks[t] = { level: labelLevel, rangeIndex: i };
+      t += range;
+    }
```

Comment:
What do you think of using a `for` loop?

`for (let t = Math.floor(minT / range) * range; t <= maxT; t += range) {
      labelMarks[t] = { level: labelLevel, rangeIndex: index };
    }`

Reply:
done